### PR TITLE
DotNet: Add support for payload to openapi skill

### PR DESCRIPTION
**Pull request by thomasgobin**
### Motivation and Context

I am adding support for payload for openapi skill.
See Issue #1423 

### Description

This PR will add the description of the payload such as the example schema and description of all properties.

### Contribution Checklist

- [X] The code builds clean without any errors or warnings
- [X] The PR follows SK Contribution Guidelines (https://github.com/microsoft/semantic-kernel/blob/main/CONTRIBUTING.md)
- [X] The code follows the .NET coding conventions (https://learn.microsoft.com/dotnet/csharp/fundamentals/coding-style/coding-conventions) verified with `dotnet format`
- [X] All unit tests pass, and I have added new tests where possible
- [X] I didn't break anyone :smile:

### CODE REVIEW

**Review by dmytrostruk — COMMENTED** [review]
Hi @thomasgobin !
Thank you for contribution and sorry for delayed response! 

I left initial comments on new changes. Also, it would be great to add some unit tests to verify new functionality and see what cases it covers. Thanks again!

File: dotnet/src/Skills/Skills.OpenAPI/Extensions/KernelOpenApiExtensions.cs [hunk 1]
```diff
@@ -295,6 +296,31 @@ async Task<SKContext> ExecuteAsync(SKContext context)
...
+        {
+            List<string> propertiesDescription = new();
+            var payloadDescription = KernelOpenApiExtensions.ExtractProperty(operation.Payload.Properties, ref propertiesDescription);
+            if (payloadDescription != null && payloadDescription.Count > 0)
+            {
+                var payloadJsonSample = JsonConvert.SerializeObject(payloadDescription, Formatting.None);
+                var payloadParameter = parameters.FirstOrDefault(x => x.Name == "payload" || x.Name == "payload");
+
+                // Aggregate JSON properties descriptions
+                var payloadJsonDescription = string.Join("\n        ", propertiesDescription);
```
[the remark this answers is not in the document]
Comment (glahaye) [reply dotnet/src/Skills/Skills.OpenAPI/Extensions/KernelOpenApiExtensions.cs:309]: Maybe use "\n\t" instead??

File: dotnet/src/Skills/Skills.OpenAPI/Extensions/KernelOpenApiExtensions.cs [hunk 2]
```diff
@@ -21,6 +21,7 @@
 using Microsoft.SemanticKernel.Skills.OpenAPI.Model;
 using Microsoft.SemanticKernel.Skills.OpenAPI.OpenApi;
 using Microsoft.SemanticKernel.Skills.OpenAPI.Skills;
+using Newtonsoft.Json;
```
Comment (glahaye) [inline dotnet/src/Skills/Skills.OpenAPI/Extensions/KernelOpenApiExtensions.cs:24]: Are we still using Newtonsoft.Json, or trying to shift over to System.Text.Json?

Comment (shawncal) [reply dotnet/src/Skills/Skills.OpenAPI/Extensions/KernelOpenApiExtensions.cs:24]: We should use System.Text.Json wherever possible. As I recall, the nuget package we call to parse OpenAPI docs currently uses newtonsoft so, unfortunately, this package brings it in.

**Comment by shawncal** [comment]
@thomasgobin Since this PR hasn't had any activity in a while, I'm going to flip it to "draft". If you're able to address the comments, please go ahead and reactivate. Otherwise, we'll close it in another week.

@SergeyMenshykh please take a look at the associated Issue for the desired functionality here, and advise on priority and potential solutions.

**Comment by lemillermicrosoft** [comment]
Thanks for your PR and contribution. For now, we are closing old drafts. Please re-open as appropriate.

**Pull request closed without merging**